### PR TITLE
Lead acid batteries [CR]

### PR DESCRIPTION
**Pull request by mugling**
Follows discussion in #17037

Now that alternators use real-world values (#17166) this PR also converts the lead-acid SLI batteries using 1 charge = 1kJ. As per @Coolthulhu's suggestion the `truck battery` is deprecated - IRL they don't exist as a discrete items with larger vehicles instead started using multiple batteries. The issue of storage batteries is beyond the scope of this PR but can be considered further in #17037.

Lead-acid batteries become much heavier, mainly due to their `lead` content with the latter now having a more appropriate density. A larger `stack_size` is chosen as it is also used for crafting ammunition. Crafting of separate battery plates is dropped as they served only as a tedious and otherwise useless intermediate step.

Finally lead-acid batteries now contain `sulphuric acid` which itself is rescaled to 1 charge = 1ml. The specific concentration of the acid is left undefined but is probably in the region of 50% which is also the upper-bound on what could spawn in a hardware store (5d4d9e9). In any case there probably is a limit to the amount of detail we need to simulate and in terms of accuracy the new crafting recipes are much better than the previous generic acid.

**tl;dr**
- for lead-acid batteries  `charges` are expressed in `kJ` and are based upon real-world examples
- `lead` is now a dense material consuming little volume but much weight
- lead-acid batteries both require and are a source of `sulphuric acid`
- marked `CR` because `truck battery` is deprecated

### CODE REVIEW

File: data/json/items/migration.json [hunk 1]
```diff
@@ -5,6 +5,11 @@
     "replace": "adv_UPS_off"
   },
   {
+    "id": "battery_truck",
+    "type": "MIGRATION",
+    "replace": "battery_car"
```
Comment (Coolthulhu) [inline data/json/items/migration.json]: Risky: there are (player-made) vehicles with truck and car batteries mounted on the same tile.

**Comment by Coolthulhu** [comment]
> Lead-acid batteries become much heavier, mainly due to their lead content with the latter now having a more appropriate density.

Car battery has old weight.

> Finally lead-acid batteries now contain sulphuric acid which itself is rescaled to 1 charge = 1ml.

Recipes that use sulphuric acid (3 of them in chem stuff) need to be changed too.

Also, the primary uses of car battery acid were crafting batteries and crafting acidbombs.
I see the following options:
* Add a very simple recipe that turns the sulphuric acid into the generic concentrated acid (probably the best for now)
* Add sulphuric acid to recipes where concentrated acid is used (possibly replacing concentrated acid)
* Drop the change for now, make them actually acid in a separate PR also handling recipes and mods

> truck battery is deprecated - IRL they don't exist as a discrete items with larger vehicles instead started using multiple batteries.

But all the relevant vehicles here only lost their truck battery, they didn't gain the second car battery.

**Comment by mugling** [comment]
> Recipes that use sulphuric acid (3 of them in chem stuff) need to be changed too.

Those recipes were both horribly inaccurate and entirely unused so were dropped

**Comment by mugling** [comment]
> Lead-acid batteries become much heavier, mainly due to their lead content with the latter now having a more appropriate density.
> 
> Car battery has old weight.

Ah yes, it was late when I was writing that. The lead-acid batteries had fairly reasonable weights but their volumes changed considerably.

**Comment by mugling** [comment]
> Also, the primary uses of car battery acid were crafting batteries and crafting acidbombs.

The former is a silly recipe as car batteries != alkaline cells, the latter could use sulphuric battery acid with ease although there is talk in a separate PR of dropping acidbombs entirely?

File: data/json/items/migration.json (hunk 1 above)
Comment (mugling) [inline data/json/items/migration.json]: Is that per-se a problem beyond once you remove one of the doubled up batteries you can't reinstall it? Eventually I want to support multiple parts on a vehicle tile (#17062).

**Comment by Coolthulhu** [comment]
> The former is a silly recipe as car batteries != alkaline cells

We have lemon and vinegar batteries. Dropping the improvised battery recipe should be in a separate PR.

> there is talk in a separate PR of dropping acidbombs entirely

They should be supported until they're dropped.

**Comment by mugling** [comment]
> We have lemon and vinegar batteries. Dropping the improvised battery recipe should be in a separate PR.

The improvised battery recipe isn't changed by this PR?



> there is talk in a separate PR of dropping acidbombs entirely
> 
> They should be supported until they're dropped.

I can convert these to use `sulphuric acid` in this PR?

**Comment by Coolthulhu** [comment]
> The improvised battery recipe isn't changed by this PR?

But it is losing an important source of components here. Why should vinegar work better for batteries than sulphuric acid?

> I can convert these to use sulphuric acid in this PR?

Would be better off with an alternative component of sulphuric acid instead of just a replacement.

**Comment by mugling** [comment]
> But it is losing an important source of components here. Why should vinegar work better for batteries than sulphuric acid?

To be honest the improvised battery recipe is also a bit questionable

> I can convert these to use sulphuric acid in this PR?
> 
> Would be better off with an alternative component of sulphuric acid instead of just a replacement.

You lost me there

**Comment by Coolthulhu** [comment]
> To be honest the improvised battery recipe is also a bit questionable

But it's an important part of the game. It should be killed off with a separate PR.

> You lost me there

```
"components" : [
    [ [ "bottle", 1 ] ],
    [ [ "acid", 3 ], [ "sulphuric_acid", 750 ] ]
]
```
Instead of just
```
"components" : [
    [ [ "bottle", 1 ] ],
    [ [ "sulphuric_acid", 750 ] ]
]
```
or (current)
```
"components" : [
    [ [ "bottle", 1 ] ],
    [ [ "acid", 3 ] ]
]
```

**Comment by mugling** [comment]
> But it's an important part of the game

Does anyone actually craft batteries in that manner? If so I'd rather change the recipe to work with sulphuric acid for now and we can revisit that when when we redo the other batteries?

**Comment by TempestMaiden** [comment]
I used to, and if getting vehicle power up and running gets more expensive I'm likely to start again. Cracking truck batteries open can be a lot more reliable than finding vinegar.

**Comment by John-Candlebury** [comment]
Does anyone actually craft batteries in that manner? If so I'd rather change the recipe to work with sulphuric acid for now and we can revisit that when when we redo the other batteries?

I do, vinegar has few utility besides that.

**Comment by Coolthulhu** [comment]
I see no consensus.
If you reverted the generic acid to sulphric acid change, I could merge it right now. That change could then be reintroduced in a separate PR, preferably with changes to batteries and/or acid bombs.

**Comment by mugling** [comment]
> If you reverted the generic acid to sulphric acid change, I could merge it right now.

Ok, lets split this PR. Give me a few minutes

**Comment by mugling** [comment]
Done

**Comment by Coolthulhu** [comment]
The batteries need their folded volume updated. They still fold into volume 3 and 4.

**Comment by kevingranade** [comment]
Why do they even have a folded volume?

**Comment by Coolthulhu** [comment]
If they didn't, a single motorcycle battery would prevent the entire vehicle from being foldable.
Foldable electric bike sounds feasible to me.

**Comment by BorkBorkGoesTheCode** [comment]
> If they didn't, a single motorcycle battery would prevent the entire vehicle from being foldable.
 Foldable electric bike sounds feasible to me.

That sounds like a bug to me. Why shouldn't a bicycle fold if it contains unfoldable parts?

**Comment by mugling** [comment]
> Why do they even have a folded volume?

I think historically (currently?) some items folded to less than their base volume?

**Comment by kevingranade** [comment]
Sorry I was unclear, I know it's to adjust the volume, but why would a
battery reduce volume?

**Comment by Coolthulhu** [comment]
It doesn't, it's just that if it doesn't have a stated folded volume, it prevents the vehicle from being foldable.

**Comment by codemime** [comment]
> It doesn't, it's just that if it doesn't have a stated folded volume, it prevents the vehicle from being foldable.

Making batteries foldable looks like a workaround to me. They probably should be treated like "integral" parts and don't affect vehicle foldability.

**Comment by kevingranade** [comment]
OK that's bad but not as nonsensical as batteries changing volume.  It's
possible folded volume should be expressed as a difference, or at least
have a value indicating, "foldable, but unchanged volume".
